src/Umbraco.Web.BackOffice/Filters/EnsureUserPermissionForContentAttribute.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Umbraco.Core;
+using Umbraco.Core.Models;
+using Umbraco.Core.Security;
+using Umbraco.Core.Services;
+using Umbraco.Web.Actions;
+using Umbraco.Web.Common.Exceptions;
+using Umbraco.Web.Security;
+
+namespace Umbraco.Web.BackOffice.Filters
+{
+    /// <summary>
+    ///     Auth filter to check if the current user has access to the content item (by id).
+    /// </summary>
+    /// <remarks>
+    ///     This first checks if the user can access this based on their start node, and then checks node permissions
+    ///     By default the permission that is checked is browse but this can be specified in the ctor.
+    ///     NOTE: This cannot be an auth filter because that happens too soon and we don't have access to the action params.
+    /// </remarks>
+    public sealed class EnsureUserPermissionForContentAttribute : TypeFilterAttribute
+    {
+
+        /// <summary>
+        ///     This constructor will only be able to test the start node access
+        /// </summary>
+        public EnsureUserPermissionForContentAttribute(int nodeId)
```

Comment:
I don't think it is required. If the input is a parameter, we use the the overload taking a parameter, and that handle UDIs. 
The overload using int is only for known nodes, like the RecycleBin. In that case I think int is okay.

No matter what, It was not part of the attributes I migrated, so if we need it, it could be in its own PR. UDIs are not used internally.